packages/graph/hash_graph/lib/graph/src/store/postgres/query/expression/conditional.rs

```diff
@@ -41,6 +55,7 @@ impl Transpile for Expression<'_> {
             Self::Asterisk => fmt.write_char('*'),
             Self::Column(column) => column.transpile(fmt),
             Self::Parameter(index) => write!(fmt, "${index}"),
+            Self::Constant(constant) => constant.transpile(fmt),
```

Comment:
Just a clarification question: are we putting the constant into the query and not using a param?
Asking because it would mean we have to be a little cautious about `Constants` if/when we add more variants as to never have user input be put into the SQL string

Reply:
Yes, this will put the constant directly in the query (i.e. `latest_version = TRUE`). This avoids using parameters we return as `dyn ToSql`, which has to be passed to the statement after it has been prepared. I thought this would be cleaner, but I agree that we need to be careful when converting user input to an expression.

Reply:
I think I would like to have a warning in the transpile of the `Constant` enum to mention that we should never provide user-input as-is. Maybe we can find a way to disallow user-input in the Constant conversion functions outright, but not sure it's worth delving into

Reply:
I added a comment in c4360cbd98316636a12a35d01b50604a00f5b67e